### PR TITLE
Better documentation for all supported libraries

### LINKED ISSUE #125: docs: provide a list of all supported estimators
I think it can be good to have a centralized list with all supported estimators. It should provide:

* estimator name, a link to the original docs for the estimator;
* a link to additional parameters eli5.explain.. / eli5.show.. functions take for the estimator, and/or a link to the function which actually explains weights or predictions;
* a brief description of the explanation method, with a link to a more detailed description.

### PR DESCRIPTION
**Pull request by kmike**
This PR should fix #125. TODO:

* [x] scikit-learn
* [x] InvertableHashingVectorizer example
* [x] lightning
* [x] xgboost
* [x] sklearn-crfsuite
* [x] add links to tutorials from docs for all supported libraries
* [x] make sure it is documented how are explain_weights / explain_prediction dispatched
* [x] add links to "Supported Libraries" section from other docs if appropriate

### CODE REVIEW

File: docs/source/libraries/sklearn.rst [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+ELI5 supports many estimators, transformers and other components
+from scikit-learn_ library.
+
+.. _scikit-learn: https://github.com/scikit-learn/scikit-learn
+
+Additional ``explain_weights`` and ``explain_prediction`` parameters
+--------------------------------------------------------------------
+
+For all scikit-learn classifiers and regressors :func:`eli5.explain_weights`
```
Comment (lopuhin) [inline docs/source/libraries/sklearn.rst]: For all *supported*, right? We don't support them all yet, right?

File: docs/source/libraries/sklearn.rst [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+* ``vec`` is a vectorizer instance used to transform
+  raw features to the input of the classifier or regressor
+  (e.g. a fitted CountVectorizer instance); you can pass it
+  instead of ``feature_names``.
+
+* ``vectorized`` is a flag which tells eli5 if ``doc`` should be
+  passed through ``vec`` or not. By default it is False, meaning that
+  if ``vec`` is not None, ``vec.transform([doc])`` is passed to the
+  estimator. Set it to False if you're using ``vec`` to get feature names,
```
Comment (lopuhin) [inline docs/source/libraries/sklearn.rst]: I would say "if you are passing ``vec`` instead of ``feature_names``", or just "if you are passing ``vec``" - to make it clear that what matters is what you pass, not how you get the feature names.

File: docs/source/libraries/sklearn.rst [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+Reversing hashing trick
+-----------------------
+
+eli5 allows to recover feature names for HashingVectorizer_ and FeatureHasher_
+by computing hashes for the provided example data.
+:func:`eli5.explain_prediction` handles HashingVectorizer as ``vec``
+automatically; to handle HashingVectorizer_ and FeatureHasher_ for
+:func:`eli5.explain_weights` use
+:class:`~.InvertableHashingVectorizer` or
+:class:`~.FeatureUnhasher`.
```
Comment (lopuhin) [inline docs/source/libraries/sklearn.rst]: I think it would be nice to either have a short usage example here, or link to the tutorial where it is described.

File: docs/source/libraries/sklearn.rst [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+
+.. _CountVectorizer: http://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.CountVectorizer.html#sklearn.feature_extraction.text.CountVectorizer
+.. _TfIdfVectorizer: http://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.TfidfVectorizer.html#sklearn.feature_extraction.text.TfidfVectorizer
+.. _HashingVectorizer: http://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.HashingVectorizer.html#sklearn.feature_extraction.text.HashingVectorizer
+
+.. image:: ../static/char-ngrams.png
+
+Text highlighting also works if a document is vectorized using FeatureUnion_
+with at least one of CountVectorizer_, TfIdfVectorizer_ or HashingVectorizer_
+in the transformer list; features of other transformed are displayed in
```
Comment (lopuhin) [inline docs/source/libraries/sklearn.rst]: "transforme*rs*"

File: docs/source/libraries/sklearn.rst (hunk 3 above)
Comment (kmike) [inline docs/source/libraries/sklearn.rst]: Yeah, a good point. I hoped that a link to InvertableHashingVectorizer is enough (there is usage information in a docstring), but an example right here would be better indeed. I'll add it.

**Comment by kmike** [comment]
@lopuhin this is ready for your review :)
